### PR TITLE
Upgrade to Polkadot v0.9.40

### PR DESCRIPTION
**Pull request by jasl**
https://github.com/paritytech/substrate/compare/polkadot-v0.9.39...polkadot-v0.9.40

Notable upstream change:
- https://github.com/paritytech/substrate/pull/13443
- https://github.com/paritytech/substrate/pull/13458

0.9.40 no visible change, just boring renaming, deprecating, and some dependencies upgrade

### CODE REVIEW

File: crates/pink/capi/src/v1/types.rs
Comment (kvinwang) [inline crates/pink/capi/src/v1/types.rs:1]: Is this file regenerated on macOS? Let's revert it.
Or remove it from the repo.

Comment (jasl) [reply crates/pink/capi/src/v1/types.rs:1]: Should we ignore this file?

Comment (jasl) [reply crates/pink/capi/src/v1/types.rs:1]: I remove the file and add it to `.gitignore`, it generated by bindgen so will suddenly change

**Review by kvinwang — APPROVED** [review]
LGTM

**Pull request merged**